src/main/scala/org/apache/sysml/compiler/lang/source/DML.scala

```diff
@@ -0,0 +1,473 @@
+              case (Some(tgt), (arg :: Nil) :: Nil) => {
+
+                val module = tgt(env)
+
+                // methods in the package object (builtins with one argument (read)
+                if (module == "package") {
+                  printBuiltin(tgt, method, argss, env)
+                }
+
+                // matrix constructors with one argument (fromDataFrame)
+                else if (module == "Matrix") {
+                  printConstructor(method, argss, env, false)
+                }
+
+                else if (module == "Vector") {
+                  printConstructor(method, argss, env, true)
+                }
+
+                // methods from scala.predef with one argument (println(...) etc.)
+                else if (module == "Predef") {
+                  val name: u.TermName = method.name.decodedName.toTermName
+
+                  name match {
+                    case u.TermName(tn) if tn == "println" || tn == "print" => s"print(${arg(env)})"
+                    case u.TermName("intWrapper") => s"${arg(env)}"
+                    case _ => abort(s"scala.predef.$name is not supported in DML", method.pos)
+                  }
+                }
+
+                // binary operators
+                else {
+                  method.name.decodedName.toTermName match {
+                    case u.TermName("to")    => s"${tgt(env)} + 1:${arg(env)} + 1"
+                    case u.TermName("until") => s"${tgt(env)} + 1:${arg(env)}"
+                    case u.TermName("%") => s"($module %% ${args(0)})" // modulo in dml is %%
+                    case u.TermName("&&") => s"($module & ${args(0)})" // && in dml is &
+                    case u.TermName("||") => s"($module | ${args(0)})" // || in dml is |
+                    case _ => s"($module ${method.name.decodedName} ${args(0)})"
+                  }
+                }
+              }
+
+              // matches apply methods with multiple arguments
+              case (Some(tgt), (x :: xs) :: Nil) if isApply(method) => {
+                val module = tgt(env)
+                val argString = args.mkString(" ")
+
+                if (module == "Array") {
+                  // sequence/array constructors
+                  s""""$argString""""
+                }
+
+                else if (module == "Matrix") {
+                  printConstructor(method, argss, env, false)
+                }
+
+                else if (module == "Vector") {
+                  printConstructor(method, argss, env, true)
+                }
+
+                else if (module == "package") {
+                  // builtins
+                  "builtin"
+                }
+
+                  // apply as indexing - convert Scala 0-based to DML 1-based
+                else {
+                  val rows = args(0) // rows
+                  val cols = args(1) // columns
+
+                  (rows, cols) match {
+                    case (":::", c) if c.contains(":")  => s"$module[,$c]"
```

Comment:
I would have added parentheses around `$c` and `$r` in the format strings below, just in case. For example, `$c + 1` --> `($c) + 1`